contracts/liquidity_hub/pool-network/terraswap_pair/src/helpers.rs

```diff
@@ -394,12 +397,28 @@ pub fn assert_slippage_tolerance(
         let pools: [Uint256; 2] = [pools[0].amount.into(), pools[1].amount.into()];
 
         // Ensure each prices are not dropped as much as slippage tolerance rate
-        if Decimal256::from_ratio(deposits[0], deposits[1]) * one_minus_slippage_tolerance
-            > Decimal256::from_ratio(pools[0], pools[1])
-            || Decimal256::from_ratio(deposits[1], deposits[0]) * one_minus_slippage_tolerance
-                > Decimal256::from_ratio(pools[1], pools[0])
-        {
-            return Err(ContractError::MaxSlippageAssertion {});
+        match pair_type {
+            PairType::StableSwap { .. } => {
+                let pools_total = pools[0].checked_add(pools[1])?;
+                let deposits_total = deposits[0].checked_add(deposits[1])?;
+
+                let pool_ratio = Decimal256::from_ratio(pools_total, pool_token_supply);
+                let deposit_ratio = Decimal256::from_ratio(deposits_total, amount);
+
+                if pool_ratio * one_minus_slippage_tolerance > deposit_ratio {
+                    return Err(ContractError::MaxSlippageAssertion {});
+                }
+            }
+            PairType::ConstantProduct => {
+                if Decimal256::from_ratio(deposits[0], deposits[1]) * one_minus_slippage_tolerance
+                    > Decimal256::from_ratio(pools[0], pools[1])
+                    || Decimal256::from_ratio(deposits[1], deposits[0])
```

Comment:
nit: I'd love a small bit of doc describing at a high level the difference. Is it an math equation

Reply:
Good point. It has to do with the characteristics and the math of the stableswap, we can't use the formula for the constant product as that doesn't account for the amp factor used in the stableswap. I added a comment to this.